### PR TITLE
[WIP] add supercell fields to MigrationGraphDoc

### PR DESCRIPTION
**Pull request by hmlli**
*Start with a description of this PR. Then edit the list below to the items that make sense for your PR scope, and check off the boxes as you go!*

## Contributor Checklist

- [x] I have broken down my PR scope into the following TODO tasks
   -  [x] add attributes to MigrationGraphDoc
   -  [x] add static method to generate the fields
- [x] I have run the tests locally and they passed.
- [x] I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR

### CODE REVIEW

**Comment by hmlli** [comment]
@munrojm Hi Jason! This PR is pretty much ready to be merged - I would like to ask you to take a look at the warnings and deprecated field though, as I'm not sure how they're supposed to be implemented other than adding them to the doc's fields. Otherwise I think we're good to go!

**Comment by munrojm** [comment]
Hi @hmlli, looks pretty good! One last thing that I think needs changing is the field requirements. Are there any that you feel should _**always**_ be included in the doc if one is to be built? If so, I would make those required with `...` or a default value, and also make them necessary arguments one must pass to the construction method. Then, the remainder of the fields can be passed through `**kwargs` (including `warnings` and `deprecated`).

File: emmet-core/emmet/core/mobility/migrationgraph.py [hunk 1]
```diff
@@ -66,10 +115,167 @@ def from_entries_and_distance(
             max_distance=hop_cutoff
         )
 
+        host_sc, sc_mat, min_length, minmax_num_atoms, coords_dict, combo = MigrationGraphDoc.generate_sc_fields(
+            mg=migration_graph,
+            min_length=min_length,
+            minmax_num_atoms=minmax_num_atoms,
+            sm=sm
+        )
+        print(coords_dict)
```
Comment (munrojm) [inline emmet-core/emmet/core/mobility/migrationgraph.py:124]: This print statement crept in and should be removed. @hmlli

**Comment by hmlli** [comment]
@munrojm the required fields are implemented and it should be ready to merge. Please lmk if anything else needs to be changed! Thanks :)

**Comment by munrojm** [comment]
Looks good. Thanks @hmlli!

**Pull request merged**